src/runtime.rs

```diff
@@ -315,7 +315,7 @@ impl<'a> SnapshotAssertionContext<'a> {
             .canonicalize()
             .ok()
             .and_then(|s| {
-                s.strip_prefix(self.cargo_workspace.as_path())
+                s.strip_prefix(self.cargo_workspace.canonicalize().unwrap())
```

Comment:
Unfortunately `canonicalize` can fail quite commonly. Ideally the error here is swallowed.